rust/flatbuffers/src/endian_scalar.rs

```diff
@@ -150,30 +150,36 @@ pub fn byte_swap_f64(x: f64) -> f64 {
 /// endian conversion, if necessary.
 #[inline]
 pub fn emplace_scalar<T: EndianScalar>(s: &mut [u8], x: T) {
-    let sz = size_of::<T>();
-    let mut_ptr = (&mut s[..sz]).as_mut_ptr() as *mut T;
-    let val = x.to_little_endian();
+    let x_le = x.to_little_endian();
     unsafe {
-        *mut_ptr = val;
+        core::ptr::copy_nonoverlapping(
+            &x_le as *const T as *const u8,
+            s.as_mut_ptr() as *mut u8,
+            size_of::<T>()
+        );
     }
 }
```

Comment:
This is missing checking the length of `s` and still allows reading uninitialized padding bytes of structs.

Reply:
What do you mean by "still allows reading uninitialized padding bytes of structs"

Reply:
```rust
struct Struct {
    x: u8,
    y: u16,
}
```

This struct has one padding byte, which is uninitialized. Implementing `EndianScalar` for that struct would allow reading such uninitialized byte.